src/treaty2/index.ts

```diff
@@ -196,6 +196,12 @@ const createProxy = (
 							continue
 						}
 
+						if (typeof value === 'object') {
+							append(key, JSON.stringify(value))
```

Comment:
append(key, JSON.stringify(value)) this doesn't work for the date.
This append double quotas to date and the server fails the validation because it is waiting for a date 
`?startDate="2024-10-05T22:00:00.000Z"&endDate="2024-10-12T22:00:00.000Z"`
Maybe a if before that check if value === date and return isostring should work better